### PR TITLE
refactor: remove redundant propTypes in ColumnElement

### PR DESCRIPTION
**Pull request by EugeneTorap**
### SUMMARY
Refactoring ColumnElement component and remove redundant propTypes code

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/SqlLab/components/ColumnElement/index.tsx [hunk 1]
```diff
@@ -74,14 +69,14 @@ export type ColumnKeyTypeType = keyof typeof tooltipTitleMap;
 
 interface ColumnElementProps {
   column: {
-    name: string;
+    name: React.ReactNode;
```
[the remark this answers is not in the document]
Comment (EugeneTorap) [reply superset-frontend/src/SqlLab/components/ColumnElement/index.tsx]: Done

**Comment by EugeneTorap** [comment]
@geido Can you review it again?
Also review #18141, #18142, #18143

**Review by ad-m — APPROVED** [review]

**Comment by geido** [comment]
Just started the CI. I'll keep an eye

**Comment by EugeneTorap** [comment]
> Just started the CI. I'll keep an eye

Can also test 18141, 18142, 18143 PR?

**Comment by EugeneTorap** [comment]
@lyndsiWilliams @AAfghahi  Can you merge the PR?

File: superset-frontend/src/SqlLab/components/ColumnElement/index.tsx [hunk 2]
```diff
@@ -80,7 +75,7 @@ interface ColumnElementProps {
   };
 }
 
-export default function ColumnElement({ column }: ColumnElementProps) {
+const ColumnElement: FC<ColumnElementProps> = ({ column }) => {
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/ColumnElement/index.tsx:78]: I've seen instances where `React.FC` can be problematic ([example](https://github.com/typescript-cheatsheets/react#function-components)) - could you either define this more explicitly or write some tests to ensure this won't cause any issues?

**Review by lyndsiWilliams — CHANGES_REQUESTED** [review]
I've got just one question, otherwise this looks good 😁

**Comment by lyndsiWilliams** [comment]
> @lyndsiWilliams @AAfghahi Can you merge the PR?

We can review and approve PRs, but we can't merge unless there's also an approval from a higher-level engineer.

**Comment by EugeneTorap** [comment]
> > @lyndsiWilliams @AAfghahi Can you merge the PR?
> 
> We can review and approve PRs, but we can't merge unless there's also an approval from a higher-level engineer.

@lyndsiWilliams Got it, can you review it again?

**Comment by lyndsiWilliams** [comment]
> > > @lyndsiWilliams @AAfghahi Can you merge the PR?
> > 
> > 
> > We can review and approve PRs, but we can't merge unless there's also an approval from a higher-level engineer.
> 
> @lyndsiWilliams Got it, can you review it again?

The code looks good! Could you fill in the description of your PR?

**Comment by EugeneTorap** [comment]
> > > > @lyndsiWilliams @AAfghahi Can you merge the PR?
> > > 
> > > 
> > > We can review and approve PRs, but we can't merge unless there's also an approval from a higher-level engineer.
> > 
> > 
> > @lyndsiWilliams Got it, can you review it again?
> 
> The code looks good! Could you fill in the description of your PR?

Sure, done

**Review by lyndsiWilliams — APPROVED** [review]

**Review by AAfghahi — APPROVED** [review]

**Pull request merged**